engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/impl/TaskRestServiceImpl.java

```diff
@@ -159,4 +178,27 @@ public void createTask(TaskDto taskDto) {
   public TaskReportResource getTaskReportResource() {
     return new TaskReportResourceImpl(getProcessEngine());
   }
+
+  private List<TaskDto> getVariablesForTasks(ProcessEngine engine,
+                                             List<Task> matchingTasks,
+                                             boolean withTaskVariablesInReturn,
+                                             boolean withCommentAndAttachments) {
+    TaskService taskService = engine.getTaskService();
+    List<TaskDto> tasks = new ArrayList<TaskDto>();
+    for (Task task : matchingTasks) {
+      VariableMap taskVariables;
+      if (withTaskVariablesInReturn) {
```

Comment:
This loop can introduce 2 extra DB query per task instance. One to get Variables and one for the Comments. 
Did you consider if you could do it with fewer DB calls?